File: performance-service/src/performance.cpp
#include "performance.h"

PerformanceResult compute_leg(
  double distance_nm,
  double tas,
  double fuel_flow
) {
  PerformanceResult result;

  result.ete_hours = distance_nm / tas;
  result.fuel_required = result.ete_hours * fuel_flow;

  return result;
}

File: api/src/services/performance.ts
export interface ComputeLegRequest {
  distance_nm: number;
  tas: number;
  fuel_flow: number;
}

export interface ComputeLegResponse {
  ete_hours: number;
  fuel_required: number;
}

export async function computeLeg(
  data: ComputeLegRequest
): Promise<ComputeLegResponse> {
  const response = await fetch("http://localhost:8081/compute", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(data)
  });

  if (!response.ok) {
    throw new Error("Performance service failed");
  }

  return response.json();
}
